client/src/pages/Restaurant/RestaurantMap/index.js

```diff
@@ -0,0 +1,67 @@
+/*global google*/
+import React from "react";
+import { GoogleMap } from "@react-google-maps/api";
+import RestaurantMarker from "./RestaurantMarker";
+import UserMarker from "./UserMarker";
+import { useState } from "react";
+import getUserCoordinates from "../../../utils/getUserCoordinates";
+
+async function getCoordinates() {
+  const coordinates = await getUserCoordinates();
+  const latitude = coordinates.coordinates.latitude;
+  const longitude = coordinates.coordinates.longitude;
+  return [latitude, longitude];
+}
+
+const containerStyle = {
+  width: "100%",
+  height: "30vh",
+};
+
+function Map({ restaurantLatitude, restaurantLongitude }) {
+  const [map, setMap] = React.useState(null);
+
+  const mapOptions = {
+    zoomControlOptions: {
+      position: google.maps.ControlPosition.RIGHT_TOP,
+    },
+    mapTypeControl: false,
+    streetViewControl: false,
+  };
+
+  const [userCoordinates, setUserCoordinates] = useState([]);
+
+  React.useEffect(() => {
+    if (!map) return;
+    getCoordinates().then((res) => setUserCoordinates(res));
+    map.setCenter({
+      lat: (userCoordinates[0] + restaurantLatitude) / 2,
+      lng: (userCoordinates[1] + restaurantLongitude) / 2,
+    });
```

Comment:
1. Make a function `computeMidpoint` that returns the midpoint of an array of coordinates. Implement this in a new file in the utils folder. call it coordinates.js
2. No need for you to make `getCoordinates`
3. No need for userCoordinates state
4. What if the user doesn't provide their location? I think something like this would be nice:
```js
getUserCoordinates()
  .then((result) => {
    const userCoordinates = result.coordinates;
    const restaurantCoordinates = {
      longitude: restaurantLongitude,
      latitude: restaurantLatitude,
    };
    const midpoint = computeMidpoint(restaurantCoordinates, userCoordinates);
    
    map.setCenter({
      lat: midpoint.latitude,
      lng: midpoint.longitude,
    });
  })
  .catch(() => {
    // in case user doesn't provide location, default to restaurant location
    map.setCenter({
      lat: restaurantLatitude,
      lng: restaurantLongitude,
    });
  });

```